pkg/autodetect/main.go

```diff
@@ -221,6 +224,37 @@ func (b *Background) detectKafka(apiList *metav1.APIGroupList) {
 	}
 }
 
+func (b *Background) hasClusterRolePermission() bool {
+	list := &rbac.ClusterRoleBindingList{}
+	opts := []client.ListOption{}
+	err := b.clReader.List(context.Background(), list, opts...)
+	if err != nil {
+		log.WithField("errormsg", err).Debug("Failed to list cluster role bindings")
+		if strings.HasSuffix(err.Error(), "cluster scope") {
+			return false
+		}
+		// still return true if not scope issue
+		return true
+	}
+	return true
+}
+
+func (b *Background) detectClusterPermissions() {
+	if b.hasClusterRolePermission() {
+		viper.Set("has-cluster-permission", true)
+	} else {
+		watchNamespace, err := k8sutil.GetWatchNamespace()
```

Comment:
I believe this might return a comma-separated list of namespaces that the operator is watching. Not sure it has any impact in this PR, but the naming could be in the plural here if that's the case.

Reply:
In any case, I think you can just set the `has-cluster-permission` and let `k8sutil.GetWatchNamespace()` to be used by the other parts.

Reply:
ok